doc/sphinx/proof-engine/ltac.rst

```diff
@@ -1083,45 +1567,49 @@ Testing boolean expressions
+   if :token:`ident` is specified, the lemma is saved with that name; otherwise
+   the lemma is saved with the name `NAME_subproofN` where
+   `NAME` is the name of the current goal and `N` is
+   a fresh name. The auxiliary lemma is inlined in the final proof term.
 
-   This tactical is useful with tactics such as :tacn:`omega` or
-   :tacn:`discriminate` that generate huge proof terms. With that tool the user
-   can avoid the explosion at time of the Save command without having to cut
-   manually the proof in smaller lemmas.
+   This is useful with tactics such as :tacn:`omega` or
+   :tacn:`discriminate` that generate huge proof terms with many intermediate
+   goals.  It can significantly reduce peak memory use.  In most cases it doesn't
+   have a significant impact on run time.  One case in which it can reduce run time
+   is when a tactic `foo` is known to always pass type checking when it
+   succeeds, such as in reflective proofs.  In this case, the idiom
+   ":tacn:`abstract` :tacn:`exact_no_check` `foo`" will save half the type
+   checking type time compared to ":tacn:`exact` `foo`".
 
    It may be useful to generate lemmas minimal w.r.t. the assumptions they
    depend on. This can be obtained thanks to the option below.
 
-   .. warning::
+   :tacn:`abstract` is an :token:`l3_tactic`.
 
-      The abstract tactic, while very useful, still has some known
-      limitations, see https://github.com/coq/coq/issues/9146 for more
-      details. Thus we recommend using it caution in some
-      "non-standard" contexts. In particular, ``abstract`` won't
-      properly work when used inside quotations ``ltac:(...)``, or
-      if used as part of typeclass resolution, it may produce wrong
-      terms when in universe polymorphic mode.
+   .. todo PR above: what "option below"??
 
-   .. tacv:: abstract @ltac_expr using @ident
+   .. warning::
 
-      Give explicitly the name of the auxiliary lemma.
+      The abstract tactic, while very useful, still has some known
+      limitations.  See https://github.com/coq/coq/issues/9146 for more
+      details. We recommend caution when using it in some
+      "non-standard" contexts. In particular, ``abstract`` doesn't
+      work properly when used inside quotations ``ltac:(...)``.
+      If used as part of typeclass resolution, it may produce incorrect
+      terms when in polymorphic universe mode.
 
-      .. warning::
+   .. warning::
 
-         Use this feature at your own risk; explicitly named and reused subterms
-         don’t play well with asynchronous proofs.
+      Provide :token:`ident` at your own risk; explicitly named and reused subterms
+      don’t play well with asynchronous proofs.
 
-   .. tacv:: transparent_abstract @ltac_expr
+   .. tacn:: transparent_abstract @ltac_expr3
       :name: transparent_abstract
 
       Save the subproof in a transparent lemma rather than an opaque one.
```

Comment:
Given the discussion above on naming, we might want to explicitly add here that the automatically generated name uses `_subterm` rather than `_subproof`

Reply:
I see it using `instance_subterm` in Theorem foobar.  The prefix is always `instance_`?

Reply:
I see `foobar_subterm` with this example and Coq 8.11.1:
```coq
Theorem foobar : True.
transparent_abstract easy.
Show Proof.
```

Which example did you take and which version of Coq?

Reply:
test-suite/success/transparent_abstract.v

Reply:
I believe the prefix is `instance_` when `transparent_abstract` is invoked from typeclass resolution on an anonymous hole.